x-pack/solutions/observability/plugins/profiling/public/components/search_functions_input/index.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiFieldSearch } from '@elastic/eui';
+import { debounce } from 'lodash';
+import React, { useCallback, useState } from 'react';
+
+interface Props {
+  onChange: (functionName: string) => void;
+}
+
+export function SearchFunctionsInput({ onChange }: Props) {
+  const [searchQuery, setSearchQuery] = useState('');
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
To avoid that and to improve the debouncing I would suggest wrapping the onChange in a callback and the debounce in a useMemo, so something like:

```suggestion
   const debouncedOnChange = useMemo(() => debounce(onChange, 500), [onChange]);

  const handleSearchChange = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      setSearchQuery(e.target.value);
      debouncedOnChange(e.target.value);
    },
    [debouncedOnChange]
  );
```
wdyt?